syntax = "proto3";

package analytics_service;

service AnalyticsService {
  rpc ProcessEvent(AnalyticsEvent) returns (EventProcessingResponse);
  rpc GetDailyMetrics(DailyMetricsRequest) returns (DailyMetricsResponse);
  rpc GetProcessedData(ProcessedDataRequest) returns (ProcessedDataResponse); 
}

message AnalyticsEvent {
  int32 user_id = 1;
  string url = 2;
  int64 timestamp = 3;
}

message EventProcessingResponse {
  bool success = 1;
}

message DailyMetricsRequest {
  int64 start_timestamp = 1;
  int64 end_timestamp = 2;
}

message DailyMetricsResponse {
  repeated DailyMetric metrics = 1;
}

message DailyMetric {
  string url = 1;
  int32 total_visits = 2;
  float avg_session_duration = 3;
  float bounce_rate = 4;
}

message ProcessedDataRequest {
  int32 user_id = 1;
}

message ProcessedDataResponse {
  repeated DailyMetric daily_pageviews = 1;
  float average_session_duration = 2;
  float bounce_rate = 3;
}
